src/agent/stacktrace-parser/src/lib.rs

```diff
@@ -0,0 +1,432 @@
+
+use anyhow::Result;
+use libclusterfuzz::get_stack_filter;
+use regex::RegexSet;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+mod asan;
+
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct StackEntry {
+    pub line: String,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub address: Option<u64>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub function_name: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub function_offset: Option<u64>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source_file_name: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source_file_path: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source_file_line: Option<u64>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub module_path: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub module_offset: Option<u64>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct CrashLog {
+    pub text: String,
+    pub sanitizer: String,
+    pub summary: String,
+    pub fault_type: String,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub call_stack: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub full_stack_details: Vec<StackEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub full_stack_names: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack_details: Vec<StackEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub minimized_stack_function_names: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub scariness_score: Option<u32>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub scariness_description: Option<String>,
+}
+
+fn function_without_args(func: &str) -> String {
+    // TODO: This is an extremely niave approach to splitting arguments.k It
+    // doesn't handle C++ well.  As an example:
+    //
+    // This turns this:
+    // "base::internal::RunnableAdapter<void (__cdecl*)(scoped_ptr<blink::WebTaskRunner::Task,std::default_delete<blink::WebTaskRunner::Task> >)>::Run",
+    //
+    // Into this:
+    // "base::internal::RunnableAdapter<void "
+    //
+    // However, given this is intended to enable de-duplicating crash reports
+    // with ClusterFuzz, this is going to stay this way for now. This is used to
```

Comment:
Does that mean we are _inheriting_ this?

Reply:
For now, yes.  We should revisit this.